doc/stability_wasm_proposals.md

```diff
@@ -0,0 +1,73 @@
+# Wasm Proposals
+
+This document is intended to describe the current status of WebAssembly proposals and WASI proposals in WAMR.
+
+Only track proposals that are followed in the [WebAssembly proposals](https://github.com/WebAssembly/proposals) and [WASI proposals](https://github.com/WebAssembly/WASI/blob/main/Proposals.md).
+
+Normally, the document tracks proposals that are in phase 4. However, if a proposal in an earlier phase receives support, it will be added to the list below.
+
+The _status_ represents the configuration _product-mini/platforms/linux/CMakeLists.txt_. There may be minor differences between the top-level CMakeLists and platform-specific CMakeLists.
+
+Users can turn those features on or off by using compilation options. If a relevant compilation option is not available(`N/A`), it indicates that the feature is permanently enabled.
+
+## On-by-default Wasm Proposals
+
+| Proposal                              | Phase 4 | Compilation Option         |
+| ------------------------------------- | ------- | -------------------------- |
+| Non-trapping float-to-int conversions | Yes     | N/A                        |
+| Sign-extension operators              | Yes     | N/A                        |
+| Multi-value                           | Yes     | N/A                        |
+| Reference Types                       | Yes     | `WAMR_BUILD_REF_TYPES`     |
+| Bulk memory operations                | Yes     | `WAMR_BUILD_BULK_MEMORY`   |
+| Fixed-width SIMD[^1]                  | Yes     | `WAMR_BUILD_SIMD`          |
+| Extended Constant Expressions         | Yes     | N/A                        |
+| Typed Function References             | Yes     | `WAMR_BUILD_GC`            |
```

Comment:
Maybe type function references should be in off-by-default? Since it needs to enable GC first